src/values/ObjectValue.js

```diff
@@ -405,15 +405,15 @@ export default class ObjectValue extends ConcreteValue {
   }
 
   havoc(): void {
-    this._isHavoced = this.$Realm.intrinsics.true;
+    this._isLeaked = this.$Realm.intrinsics.true;
   }
 
-  mightBeHavocedObject(): boolean {
-    return this._isHavoced.mightBeTrue();
+  isLeakedObject(): boolean {
+    return this._isLeaked.equals(this.$Realm.intrinsics.true);
   }
 
-  mightNotBeHavocedObject(): boolean {
-    return this._isHavoced.mightNotBeTrue();
+  isNotLeakedObject(): boolean {
```

Comment:
Or you could cut `isNotLeakedObject` altogether, as it's really just `!isLeakedObject` now.